src/map/script.c

```diff
@@ -6602,13 +6602,13 @@ static BUILDIN(areawarp)
 		}
 	}
 
-	if( (m = map->mapname2mapid(mapname)) < 0 )
-		return true;
+	if ((m = script->mapname2mapid(st, mapname)) < 0)
+		return false;
 
 	if( strcmp(str,"Random") == 0 )
 		index = 0;
-	else if( !(index=script->mapindexname2id(st,str)) )
-		return true;
+	else if (!(index = script->mapindexname2id(st, str)))
```

Comment:
since you're fixing it anyway, could you change the ```!``` into ```== 0``` check? boolean operations on integers are not okay, this comment can be ignored, just an optional thing to do if you want.